web/src/main/java/by/academy/jee/web/servlet/AddTeacherServlet.java

```diff
@@ -0,0 +1,80 @@
+package by.academy.jee.web.servlet;
+
+import by.academy.jee.model.person.Person;
+import by.academy.jee.model.person.Teacher;
+import by.academy.jee.util.Initializer;
+import by.academy.jee.util.PasswordHasher;
+import by.academy.jee.util.SalaryGenerator;
+import by.academy.jee.web.util.SessionUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Map;
+
+import static by.academy.jee.web.constant.Constant.*;
+
+@WebServlet(value = "/addTeacher")
+public class AddTeacherServlet extends HttpServlet {
+
+    private static final Logger log = LoggerFactory.getLogger(AddTeacherServlet.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        SessionUtil.setupForward(this, req, resp, ADD_TEACHER_JSP_URL);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+        String userName = req.getParameter("login");
+        String password = req.getParameter("password");
+        String fio = req.getParameter("userName");
+        String ageString = req.getParameter("age");
+        String minSalaryString = req.getParameter("minSalary");
+        String maxSalaryString = req.getParameter("maxSalary");
+        Person user = Initializer.adminDao.read(userName); // TODO - add check in studentDao
+        if (user == null) {
+            user = Initializer.teacherDao.read(userName);
+        }
+        if (user != null) {
+            log.info("Error - attempt to add already existed user {}", userName);
+            req.setAttribute(ERROR_MESSAGE, USER_IS_ALREADY_EXIST);
+            SessionUtil.setupForward(this, req, resp, ADD_TEACHER_JSP_URL);
+            return;
+        }
+        int age;
+        double minSalary, maxSalary;
+        try {
+            age = Integer.parseInt(ageString);
+            minSalary = Double.parseDouble(minSalaryString);
+            maxSalary = Double.parseDouble(maxSalaryString);
+        } catch (NumberFormatException e) {
+            log.info("Error - wrong numbers format");
```

Comment:
Тут лучше не info, а error использовать
```suggestion
            log.error("Error - wrong numbers format");
```
Ну, и в аналогичных случаях тоже.

Reply:
Исправил. Хотя мне казалось, что error используется в случаях, когда может лечь приложение, а здесь по сути идет безопасная обработка

Reply:
Обычно подход такой: если сценарий не может дойти до конца, то это error.